src/packages/core/src/servers/http-server.ts

```diff
@@ -93,10 +94,57 @@ function sendResponse(
   });
 }
 
-export type HttpServerOptions = Pick<InternalOptions["server"], "rpcEndpoint">;
+function sendChunkedResponse(
+  response: HttpResponse,
+  closeConnection: boolean,
+  statusCode: HttpResponseCodes,
+  contentType: RecognizedString | null,
+  data: Generator<Buffer, any, unknown>,
+  chunkSize: number,
+  writeHeaders: (response: HttpResponse) => void = noop
+) {
+  const { value: first } = data.next();
+  const fragments = getFragmentGenerator(data, first, chunkSize);
+  // get our first fragment
+  const { value: firstFragment } = fragments.next();
+  // check if there is any more fragments after this one
+  let { value: prevFragment, done } = fragments.next();
```

Comment:
might want to add a `.hasMore` method here that doesn't actually create the next fragment, but does determine if it _will_ has more. (maybe just make an issue for this)